testplan/common/report/base.py

```diff
@@ -52,7 +52,8 @@ def __init__(self,
                  description=None,
                  uid=None,
                  entries=None,
-                 parent_uids=None):
+                 parent_uids=None,
+                 xfail=None):
```

Comment:
I don't think it is the right design to have a dedicated xfail attribute in the report. The report already has a status field, we just need to add XFAIL as a new status and add logic into MultiTest in order to set it correctly.

Reply:
Oh OK I think I understand why you added this now. It is not just a simple boolean flag but contains information on the xfail reason and whether it was strict or not?

For the reason, I wonder if this can be done in a more generic way by adding a "reason" attribute to the status object or a `status_reason` attribute to the report object? That way when user marks a testcase as skipped, it can reuse the same reason attribute instead of adding another `skip` attribute (could also be used to give optional reason for marking testcase as ERROR etc).

As for the `strict` param, I'm not sure the report needs to care about that.

Reply:
One more thing to note as well. The current design is that the report base classes under `testplan/common/report` are super generic and know nothing about _test_ reports, which are covered by the `testplan/report/testing`. So if we _were_ to add information to the report that is specific to xfail, it should go into one or more of the classes under `testplan/report/testing` and not the common base classes.

With that said, I personally do not really see much reason currently for splitting the report classes between the generic "common" location and the test-report specific locations, since there are no other report types that inherit from the common base classes - I think it over complicates things for no real gain. So I would be happy to move all of `testplan/common/report` into `testplan/report/testing` and simplify the inheritance hierarchy as it currently very complex.